service/controller/resource/nodeautorepair/create.go

```diff
@@ -0,0 +1,116 @@
+package nodeautorepair
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/giantswarm/badnodedetector/pkg/detector"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/giantswarm/aws-operator/service/controller/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/key"
+)
+
+const (
+	annotationEnableNodeTermination = "node-auto-repair.giantswarm.io"
+
+	nodeTerminationTickThreshold = 6
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	var err error
+	cr, err := key.ToCluster(ctx, obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// check for annotation enabling the node auto repair feature
+	if _, ok := cr.Annotations[annotationEnableNodeTermination]; !ok {
+		r.logger.LogCtx(ctx, "level", "debug", "message", " node auto repair is not enabled for this cluster, cancelling")
+		return nil
+	}
+
+	if cc.Client.TenantCluster.K8s == nil {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "kubernetes clients are not available in controller context yet")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+
+		return nil
+	}
+
+	var detectorService *detector.Detector
+	{
+		detectorConfig := detector.Config{
+			K8sClient: cc.Client.TenantCluster.K8s.CtrlClient(),
+			Logger:    r.logger,
+
+			LockName:              Name,
```

Comment:
How does locking work again? I feel like I missed this in the other PRs. 😬

Reply:
Oh god I am an idiot I forgot to request for review that one

https://github.com/giantswarm/badnodedetector/pull/2

:see_no_evil:

Reply:
removed the lock fro the lib and added used kubelock instead